test/e2e/specs/default-config/recordedit/add-defaults.spec.js

```diff
@@ -200,3 +216,48 @@ describe('Record Add with defaults', function() {
         });
     });
 });
+
+describe("Record Edit with immutable columns", function() {
+
+    describe("should verify the presentation of data", function () {
+
+        beforeAll(function () {
+            var keys = [];
+            keys.push(testParams.edit_key.name + testParams.edit_key.operator + testParams.edit_key.value);
+            browser.ignoreSynchronization=true;
+            browser.get(browser.params.url + "/recordedit/#" + browser.params.catalogId + "/defaults:" + testParams.table_name + "/"  + keys.join("&"));
+
+            chaisePage.waitForElement(element(by.id("submit-record-button")));
+        });
+
+        for (var i=0; i < testParams.re_column_names.length; i++) {
+            (function(columnName) {
+                if (columnName == "foreign_key_disabled") {
```

Comment:
I think using switch case is better than if else here.

Reply:
👍